ignite/distributed/launcher.py

```diff
@@ -229,7 +229,6 @@ def __init__(
         self.backend = backend
         self._spawn_params = None
         self.init_method = init_method
-        self.logger = setup_logger(__name__ + "." + self.__class__.__name__, distributed_rank=0)
         # distributed_rank=0 <=> explicit rank 0, avoid call idist. Critical for TPU on Colab, avoid context setup
```

Comment:
Do you respect this comment also for TPUs ? 
We can remove the comment by the way

Reply:
`idist` doesn't work on colab ? I completely missed that.

Btw, this setup should be exclusively done on colab, because it had side effect.
```python
parallel = idist.Parallel(backend="gloo", nproc_per_node=1)
parallel.logger.info("who print this ?")
```

I would prefer remove `self.logger` and catch a fresh one when we need to write something.

Reply:
Almost all idist functions work on colab but there are few restrictions to keep in mind when you set up the dist model: if we call `xm.*` methods in a single proc, it will initialize single tpu device context and wont be able to init 8 tpus with multiprocessing.

Reply:
I think we should also forbid the usage of `idist.Parallel` without context manager as in your example.

Reply:
> Almost all idist functions work on colab but there are few restrictions to keep in mind when you set up the dist model: if we call `xm.*` methods in a single proc, it will initialize single tpu device context and wont be able to init 8 tpus with multiprocessing.

It sounds like as we should not use loggers (mostly internally) before process group initialization.